### DISCUSSION TITLE
Pub.dev

### DISCUSSION THREAD

**Discussion by nohli** [ali-thowfeek/simple_html_css_flutter#14] [Q&A] [answered]
Hi! I've updated the nnbd branch. It can be published to pub.dev and will appear as a preview (like how the prerelease is shown right now). After the next Flutter update (which is very close), the preview will automatically replace the stable version.

https://medium.com/dartlang/preparing-the-dart-and-flutter-ecosystem-for-null-safety-e550ce72c010

(I did the same: https://pub.dev/packages/hue_rotation/versions/1.0.0)

**Comment by ali-thowfeek** [comment, marked as answer]
I've published the preview 3.0.0 thank you for the support.

There were two warnings. I hope they are not big of an issue. 

> Packages dependent on a pre-release of another package should themselves be published as a pre-release version. If this package needs html_unescape version 2.0.0-nullsafety.0, coape version 2.0.0-nullsafety.0, consider publishing the package as a pre-release instead.

**Reply by nohli** [reply]
It has 110 pub points, seems to be ok for now 🙂

**Reply by nohli** [reply]
FYI Flutter 2.0.0 just got released and the preview version of this package is now the (only) stable: https://pub.dev/packages/simple_html_css

**Reply by nohli** [reply]
Also, I just updated all packages und fixed analyzer infos. Maybe this can be published.

Furthermore, what should we do about the branches? NNBD is now the stable version. Maybe we could rename it to main (https://github.com/github/renaming) and make it the default branch?

About the tests...probably you are more into this package than me...do you already know what should be tested (then I can try putting it to code)?

**Reply by ali-thowfeek** [reply]
I'll look into publishing the latest changes.

Yeah renaming the branch makes sense. I guess it would be a nightmare if we were to merge.

And I'll look into the tests.
Thanks.

**Reply by ali-thowfeek** [reply]
@nohli I've changed the default branch as you suggested. 

I noticed that you've put `sdk: '>=2.12.0-0 <3.0.0'` where the **"-0"** makes it a pre release right? So, i removed that part and made it the stable version of dart. like so, in 29f9fba I hope this change wont bring about anything unintentional.

**Reply by nohli** [reply]
Had it like that before and then changed it because of https://github.com/dart-lang/pub-dev/issues/4582.
(also updated my own two packages today with -0 and everything is fine, without -0 it was 'awaiting' for quite some time)

But the last comment in the issue indicates that it might work without -0.

The dependency doesn't make it a prerelease...you could also write >2.11.0 (probably the same), would that make it more stable? :P
If you put -something to the package version, then it would be a prerelease.

**Reply by ali-thowfeek** [reply]
oh, thank you for that. But i've published it already, sorry 😣. I guess we shall give them some time, and it would eventually be ok.
update: it seems to be ok now.

**Reply by nohli** [reply]
Great!
(just downgraded because I didn't want to screw up the score for a few days...if it's working with the stable sdk dependency, it's even better)
